### PR TITLE
Added fix for wrong application of height for herringbone-helical gears.

### PR DESCRIPTION
**Pull request by GarryBGoode**
Added fix for wrong application of height for herringbone-helical gears.

### CODE REVIEW

File: src/gggears/gggears_wrapper.py [hunk 1]
```diff
@@ -640,12 +640,23 @@ def __init__(
         )
         # correct for herringbone design
         if herringbone:
-            self.gearcore.z_vals = np.array([0, self.height / 2, self.height])
+            # assuming z_vals[0] is 0, as it should be
```
Comment (x0pherl) [inline src/gggears/gggears_wrapper.py:643]: This comment makes me nervous; should there be an exception raised if it's not zero instead?

**Review by x0pherl — APPROVED** [review]
I'm not deeply familiar with the code, but in a quick functional test of both herringbone=True and herringbone=False I ended up with a part with the correct height; which is an improvement.

I left a small suggestion based on the comment, but it's not a reason to reject the change.

I would also suggest that adding unit testing to catch that the returned height is correct might be useful. I know you mentioned that you might be adding helper methods for this -- given that it's a bugfix it would be understandable to wait, but I thought I should call it out.

File: src/gggears/gggears_wrapper.py (hunk 1 above)
Comment (GarryBGoode) [reply src/gggears/gggears_wrapper.py:643]: I'm not raising exceptions anywhere in the code, I'm more of a subscriber of the "We're all adults here" pythonism.
I might change my mind later but this is not a good place to start adding exceptions.

Though you're right, I'll look into it to make a more robust solution and consider the min/max of zvals. And I'll add a test.

**Comment by GarryBGoode** [comment]
There was 1 error from OCP instability out of 4000 tests. The OCP / CAD construction method is not the focus or scope of this branch, just the new tests added revealed the issue. 

Closing this PR despite the failure, further work is planned on CAD construction robustness.

**Pull request merged**